### PR TITLE
Download all: Add Metadata download

### PR DESCRIPTION
**Pull request by JeLuF**
image
- Popup to prevent accidental triggering of an image download
- Add ZIP download with JSON and folder support
- Use FileSaver polyfill for better browser support

https://discord.com/channels/1014774730907209781/1014774732018683928/1083307110361743441

### CODE REVIEW

File: ui/media/js/main.js [hunk 1]
```diff
@@ -354,15 +356,27 @@ function onUseAsInputClick(req, img) {
...
     const imageInferenceSteps = img.getAttribute('data-steps')
     const imageGuidanceScale = img.getAttribute('data-guidance')
+    
+    return createFileName(imagePrompt, imageSeed, imageInferenceSteps, imageGuidanceScale, suffix)
+}
+
+function onDownloadJSONClick(req, img) {
+    const jsonDownload = document.createElement('a')
+    jsonDownload.download = getDownloadFilename(img, 'json')
+    jsonDownload.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(JSON.stringify(req, null, " ")))
```
Comment (cmdr2) [inline ui/media/js/main.js:371]: Can you please check if this should be `req.reqBody`? I'm not sure, but would be good to confirm that we're encoding the correct JSON object.

**Comment by JeLuF** [comment]
image

File: ui/media/js/main.js (hunk 1 above)
Comment (JeLuF) [reply ui/media/js/main.js:372]: The `req` object in the context of the button handler is the reqBody object, cf. https://github.com/cmdr2/stable-diffusion-ui/blob/beta/ui/media/js/main.js#L305:
```py
const req = Object.assign({}, reqBody, {
                seed: result?.seed || reqBody.seed
            })
```

**Comment by cmdr2** [comment]
Thanks!

**Pull request merged**